### PR TITLE
Do not load sticky queue if not loaded

### PR DESCRIPTION
**Pull request by dnr**
**What changed?**
Restore a small optimization that got lost in some refactoring: if we get a workflow task for a sticky queue that's not loaded because it hasn't had pollers in the idle timeout, we don't have to load the queue at all. Note that we would return StickyWorkerUnavailable in any case, since HasPollerAfter would return false.

**Why?**
small optimization

**How did you test it?**
new unit test

### CODE REVIEW

File: service/matching/matching_engine.go [hunk 1]
```diff
@@ -320,10 +320,15 @@ func (e *matchingEngineImpl) AddWorkflowTask(
 		return false, err
 	}
 
-	baseTqm, err := e.getTaskQueueManager(ctx, origTaskQueue, stickyInfo, true)
+	sticky := stickyInfo.kind == enumspb.TASK_QUEUE_KIND_STICKY
+	// do not load sticky task queue if it is not already loaded, which means it has no poller.
```
[the remark this answers is not in the document]
Comment (MichaelSnowden) [reply service/matching/matching_engine.go:324]: Also, which part is performing this optimization. Is it the fact that we're passing the `sticky` flag to `getTaskQueueManager` or is it the condition on 328?

**Review by MichaelSnowden — COMMENTED** [review]
> Restore a small optimization that got lost in some refactoring

What refactoring did this get lost in? I want to understand how we missed this

File: service/matching/matching_engine.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dnr) [reply service/matching/matching_engine.go:324]: I moved the comment to the part that implements the thing described in the comment.

The `!sticky` part plus `baseTqm == nil` avoids loading, and the `!baseTqm.hasPollerAfter` is the other part, for when it's loaded but there are no pollers.

**Comment by dnr** [comment]
> What refactoring did this get lost in? I want to understand how we missed this

It was in #4657. There wasn't really a test for it, so I added one.

When I wrote it I think I was thinking "we always need to get user data to direct this task, so we should always load," but later I realized that's not true for sticky since they're not versioned themselves.

**Review by MichaelSnowden — APPROVED** [review]

**Pull request merged**